lib/sshutils/server.go

```diff
@@ -188,56 +195,90 @@ func (s *Server) Addr() string {
 	return s.listener.Addr().String()
 }
 
+func (s *Server) isClosed() bool {
+	s.RLock()
+	defer s.RUnlock()
+	return s.listenerClosed
+}
+
+func (s *Server) Serve(listener net.Listener) error {
```

Comment:
Doesn't look like you are using `mulitplexer.Mux` yet right? I didn't see it anywhere in the PR. Are you holding switching over for 2.4.1?

Reply:
this is just a first part of a PR, will be used after you release 2.4.0